File: AtlasWeb/Core/Models/DocumentDefinitionViewModel.cs
using System;
using System.Collections.Generic;
using TTObjectClasses;

namespace Core.Models
{
    public class DocumentDefinitionViewModel
    {
        public List<Tab> tabs { get; set; }
        public List<DocumentDefinitionItem> documentDefinitions { get; set; }
    }

    public class DocumentDefinitionItem
    {
        public Guid id
        {
            get;
            set;
        }

        public string name
        {
            get;
            set;
        }

        public string caption
        {
            get;
            set;
        }

        public Guid? parentID
        {
            get;
            set;
        }

        public List<DocumentDefinitionItem> items
        {
            get;
            set;
        }
    }

    public class Tab
    {
        public Guid id
        {
            get;
            set;
        }

        public string text
        {
            get;
            set;
        }
    }
}

File: AtlasWeb/Core/wwwroot/app/DocumentDefinition/Models/DocumentDefinitionViewModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'NebulaClient/Mscorlib/Guid';

export class DocumentDefinitionViewModel extends BaseModel {
    public tabs: Array<Tab>;
    public documentDefinitions: Array<DocumentDefinitionItem>;
}

export class DocumentDefinitionItem {
    @Type(() => Guid)
    id: Guid;
    name: string;
    caption: string;
    @Type(() => Guid)
    parentID: Guid;
    items?: DocumentDefinitionItem[];
}
export class Tab {
    @Type(() => Guid)
    id: Guid;
    text: string;
}